src/testWithSpringBoot_2_4/java/org/openrewrite/java/spring/boot2/HeadersConfigurerLambdaDslTest.java

```diff
@@ -38,37 +46,154 @@ void simpleContentSecurityPolicy() {
+                  @Override
+                  protected void configure(HttpSecurity http) throws Exception {
+                      http
+                              .headers()
+                              .xssProtection(protection -> protection.and());
+                  }
+              }
+              """
+          )
+        );
+    }
 
-    @Override
-    protected void configure(HttpSecurity http) throws Exception {
-        http
-                .headers(headers -> headers
-                        .contentSecurityPolicy(policy -> policy
-                                .policyDirectives("foobar")));
+    @Test
+    void xssProtectionDisable() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import org.springframework.context.annotation.Configuration;
+              import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+              import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+
+              @Configuration
+              public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+                  @Override
+                  protected void configure(HttpSecurity http) throws Exception {
+                      http
+                        .headers()
+                        .xssProtection().xssProtectionEnabled(false);
+                  }
+              }
+              """,
+            """
+              import org.springframework.context.annotation.Configuration;
+              import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+              import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+
+              @Configuration
+              public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+                  @Override
+                  protected void configure(HttpSecurity http) throws Exception {
+                      http
+                              .headers()
+                              .xssProtection(protection -> protection.disable());
+                  }
+              }
+              """
+          )
+        );
     }
-}
+
+    @Test
+    void xssProtectionHeaderValue() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import org.springframework.context.annotation.Configuration;
+              import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+              import org.springframework.security.web.header.writers.XXssProtectionHeaderWriter;
+
+              @Configuration
+              public class WebSecurityConfig {
+                  protected void configure(HttpSecurity http) throws Exception {
+                      http
+                          .headers()
+                          .xssProtection()
+                          .headerValue(XXssProtectionHeaderWriter.HeaderValue.DISABLED);
```

Comment:
When I add this line here
```suggestion
                          .xssProtection()
                          .xssProtectionEnabled(true)
                          .headerValue(XXssProtectionHeaderWriter.HeaderValue.DISABLED);
```
Then replacement ends up containing
```
diff --git a/WebSecurityConfig.java b/WebSecurityConfig.java
index 08c2b14..16e345b 100644
--- a/WebSecurityConfig.java
+++ b/WebSecurityConfig.java
@@ -7,7 +7,8 @@ 
     protected void configure(HttpSecurity http) throws Exception {
         http
                 .headers()
-                .xssProtection(protection -> protection
+                .xssProtection(protection -> protection
+                        .xssProtectionEnabled(true)
                         .headerValue(XXssProtectionHeaderWriter.HeaderValue.DISABLED));
     }
 }
```
Whereas that `xssProtectionEnabled(true)` should be removed right? Or replaced with `.and()` ?

Reply:
Yes. Forgot to test the invocation chain. Made a little fix